### DISCUSSION TITLE
How do I send hashtags to mastodon so that they can be hyperlinked and found in a hastag search?

### DISCUSSION THREAD

**Discussion by astridx** [mastodon/mastodon#23139] [Q&A] [answered]
How do I send hashtags to mastodon so that they can be hyperlinked and found in a hastag search?

I have got data like this:

```
{
   "@context":"https:\/\/www.w3.org\/ns\/activitystreams",
   "id":"https:\/\/example.org&id=63c3f6255a1f1",
   "type":"Update",
   "actor":"https:\/\/example.org\/users/test1",
   "to":[
      "https:\/\/www.w3.org\/ns\/activitystreams#Public"
   ],
   "cc":[
      "https:\/\/mastodon.social\/users\/test2"
   ],
   "object":{
      "type":"Note",

      "content":"lorum #examplehastag ipsum"
   }
}
```

If I send this to an Mastodon `inbox`, the Hashtag `#examplehastag` in the `content property` is treated like normal text.

What do I have to do to make Mastodon treat it like a hastag entered via the Mastodon GUI. What do I have to do to have it tagged with a link (tags/examplehastag) and found via Mastodon search.

I have tried to add the link myself. But a relative link is being cut out.

I cannot give an absolute link because there are various Mastodon instances.

**Comment by trwnh** [comment, marked as answer]
the `Note` should have a `tag` property containing an array that includes a node with the following properties:

```yaml
type: Hashtag
name: examplehashtag
```

typically you will also have `href: <absolute url>` as well. it should be an absolute link *despite* what you think -- the link is hijacked and replaced by mastodon to point to the local url. relative links are generally not supported by mastodon's JSON-LD processing

**Comment by mahomedalid** [comment]
I have a related question. I am sending this note to mastodon followers enclosed in a create activity:
```
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "id": "https://maho.dev/socialweb/notes/feba4590a4370b9c89cb0e570bd9d59b",
  "type": "Note",
  "content": "Navigating Risk: Lessons from Chernobyl for Software Development\u003Cbr\u003E\u003Cbr\u003EI have always found Chernobyl fascinating, it is a captivating tale of caution yes, but also a profound tale of wisdom.\nChernobyl by Midjourney\nChernobyl stands as a catalyst of chaos and a tragic human disaster, yet it also embodies the essence of ambition and reveals a perplexing paradox in its furnace of ambition.\nA few years ago, I was enthralled by the HBO miniseries that portrayed the catastrophic nuclear incident in Pripyat, Ukraine, a part of the Soviet Union at the time.\u003Cbr\u003E\u003Cbr\u003EFull article by \u003Ca href=\u0022https://hachyderm.io/@mapache/\u0022 class=\u0022u-url mention\u0022\u003E@\u003Cspan\u003Emapache\u003C/span\u003E\u003C/a\u003E: \u003Ca href=\u0027https://maho.dev/2023/09/navigating-risk-lessons-from-chernobyl-for-software-development/\u0027\u003Ehttps://maho.dev/2023/09/navigating-risk-lessons-from-chernobyl-for-software-development/\u003C/a\u003E\u003Cbr\u003E\u003Cbr\u003E \u003Ca href =\u0022https://maho.dev/tags/software-development\u0022 class=\u0022mention hashtag\u0022 rel=\u0022tag\u0022\u003E#\u003Cspan\u003Esoftware-development\u003C/span\u003E\u003C/a\u003E \u003Ca href =\u0022https://maho.dev/tags/risks\u0022 class=\u0022mention hashtag\u0022 rel=\u0022tag\u0022\u003E#\u003Cspan\u003Erisks\u003C/span\u003E\u003C/a\u003E",
  "url": "https://maho.dev/2023/09/navigating-risk-lessons-from-chernobyl-for-software-development/",
  "attributedTo": "https://maho.dev/@blog",
  "to": [
    "https://www.w3.org/ns/activitystreams#Public"
  ],
  "cc": [],
  "published": "2023-09-23T21:51:40-07:00",
  "tag": [
    {
      "Type": "Mention",
      "Href": "https://hachyderm.io/users/mapache",
      "Name": "@mapache@hachyderm.io"
    },
    {
      "Type": "Hashtag",
      "Href": "https://maho.dev/tags/software-development",
      "Name": "#software-development"
    },
    {
      "Type": "Hashtag",
      "Href": "https://hachyderm.io/tags/risks",
      "Name": "#risks"
    }
  ],
  "replies": {
    "id": "https://maho.dev/socialweb/replies/feba4590a4370b9c89cb0e570bd9d59b",
    "type": "Collection",
    "first": {
      "type": "CollectionPage",
      "next": "https://maho.dev/socialweb/replies/feba4590a4370b9c89cb0e570bd9d59b?page=true",
      "partOf": "https://maho.dev/socialweb/replies/feba4590a4370b9c89cb0e570bd9d59b",
      "items": []
    }
  }
}
```

It shows nice in mastodon, the card renders, the hashtags are clickable, but seems the hashtags are not indexed. The post does not show in any of the tags lists of the mastodon instances receiving the message.

![image](https://github.com/mastodon/mastodon/assets/952392/66ad9564-08d0-4c98-b2bd-4eaa88f4500a)

**Reply by mahomedalid** [reply]
In my case it may had to do with https://github.com/mastodon/mastodon/blob/c07028b2fae4dc692570df84227023ca90443288/app/services/fan_out_on_write_service.rb#L171

**Comment by SmartmanApps** [comment]
I'm not sure this is directly related, but I was asked to add my own experience with this, so here it is...

I experimented previously and made a post on Lemmy with hashtags in it, hoping the hashtags would get indexed when boosted by the community's bot into Mastodon, and they weren't. For the hashtags to work I have to make the post on Mastodon in the first place, so what I do now is make the post on Lemmy, then make a post on Mastodon with a link to the Lemmy post, and add the hashtags there. Same limitation applies to editing a post - you can't edit in hashtags afterwards and still have them get indexed - they only get indexed when you originally post (so if you REALLY want to add a hashtag then you just need to make the post again from scratch and not forget it this time). I posted about that at [Mastodon - Hashtag reach and readability ](https://dotnet.social/@SmartmanApps/111050458369565821)
